datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -801,6 +803,16 @@ impl DisplayAs for AggregateExec {
                 }
             }
             DisplayFormatType::TreeRender => {
+                let format_expr_with_alias =
+                    |(e, alias): &(Arc<dyn PhysicalExpr>, String)| -> String {
+                        let expr_sql = fmt_sql(e.as_ref()).to_string();
```

Comment:
I think you can avoid a string copy here like:
```suggestion
                        let expr_sql = fmt_sql(e.as_ref());
```

And then directly wrote them to `f` rather than creating strings

However I don't think this is performance critical code so it is also ok this way too (and it follows the previous pattern)

Maybe as a follow on PR we can avoid as many copies in this code

Reply:
We have to construct a string to compare with alias.